sycl/doc/extensions/proposed/sycl_ext_oneapi_named_sub_group_sizes.asciidoc

```diff
@@ -0,0 +1,301 @@
+values like 0 (which is otherwise not a meaningful sub-group size) or -1
+(which would otherwise correspond to a sub-group size so large it is unlikely
+any device would support it).
+
+|===
+|Property|Description
+
+|`sub_group_size_primary`
+|The `sub_group_size_primary` property adds the requirement that the kernel
+ must be compiled and executed with the primary sub-group size of the device to
+ which the kernel is submitted (as reported by the
+ `info::device::primary_sub_group_size` query).
+
+|`sub_group_size_automatic`
+|The `sub_group_size_automatic` property adds the requirement that the kernel
+ can be compiled and executed with any of the valid sub-group sizes associated
+ with the device to which the kernel is submitted (as reported by the
+ `info::device::sub_group_sizes` query). The manner in which the sub-group size
+ is selected is implementation-defined.
+
+|===
+
+At most one of the `sub_group_size`, `sub_group_size_primary` and
+`sub_group_size_automatic` properties may be associated with a kernel or
+device function.
+
+NOTE: No special handling is required to detect this case, since
+`sub_group_size_primary` and `sub_group_size_automatic` are simply named
+shorthands for properties associated with `sub_group_size_key`.
+
+There are special requirements whenever a device function defined in one
+translation unit makes a call to a device function that is defined in a second
+translation unit. In such a case, the second device function is always declared
+using `SYCL_EXTERNAL`. If the kernel calling these device functions is defined
+using a sub-group size property, the functions declared using `SYCL_EXTERNAL`
+must be similarly decorated to ensure that the same sub-group size is used.
+This decoration must exist in both the translation unit making the call and
+also in the translation unit that defines the function. If the sub-group size
+property is missing in the translation unit that makes the call, or if the
+sub-group size of the called function does not match the sub-group size of the
+calling function, the program is ill-formed and the compiler must raise a
+diagnostic.
+
+Note that a compiler may choose a different sub-group size for each kernel and
+`SYCL_EXTERNAL` function using an automatic sub-group size. If kernels with an
+automatic sub-group size call `SYCL_EXTERNAL` functions using an automatic
+sub-group size, the program may be ill-formed. The behavior when
+`SYCL_EXTERNAL` is used in conjunction with an automatic sub-group size is
+implementation-defined, and code relying on specific behavior should not be
+expected to be portable across implementations. If a kernel calls a
+`SYCL_EXTERNAL` function with an incompatible sub-group size, the compiler must
+raise a diagnostic -- it is expected that this diagnostic will be raised during
+link-time, since this is the first time the compiler will see both translation
+units together.
+
+
+=== Compiler flags
```

Comment:
This should not be a normative section of the specification.  Other implementations should be allowed to choose their own command line switches.  Maybe retitle the section "DPC++ compiler flags" and start with these sentences:

> This non-normative section describes command line flags that the DPC++ compiler supports.  Other compilers are free to provide their own command line flags (if any).

Reply:
Fixed in b2585f5ae486312722231c3f9ba56c38a11e612f.